File: RepoTransBench/target_projects/Python/C++/rbarrois_xworkflows/public_tests/test_public_utils.cpp
#include <gtest/gtest.h>
#include "xworkflows/utils.h"

TEST(PublicUtils, IterclassTraversalDifferent) {
    struct C {
        static constexpr int x = 10;
    };
    struct D : public C {
        static constexpr int y = 20;
    };

    auto result = xworkflows::utils::iterclass<D>();
    ASSERT_EQ(result["x"], 10);
    ASSERT_EQ(result["y"], 20);
}

TEST(PublicUtils, IterclassOverridesDifferent) {
    struct C {
        static constexpr int alpha = 7;
    };
    struct D : public C {
        static constexpr int alpha = 42;
    };

    auto result = xworkflows::utils::iterclass<D>();
    ASSERT_EQ(result["alpha"], 42);
}

File: RepoTransBench/target_projects/Python/Rust/rbarrois_xworkflows/public_tests/test_public_utils.rs
use rbarrois_xworkflows::utils;

#[test]
fn test_iterclass_traversal_different() {
    struct C {}
    struct D {}
    let result = utils::iterclass(&D{});
    assert_eq!(result["x"], 10);
    assert_eq!(result["y"], 20);
}

#[test]
fn test_iterclass_overrides_different() {
    struct C {}
    struct D {}
    let result = utils::iterclass(&D{});
    assert_eq!(result["alpha"], 42);
}